### ISSUE TITLE
Consider upgrading `rust-bitcoinconsensus` dependency

### ISSUE THREAD

**Issue by tcharding** [rust-bitcoin/rust-bitcoin#2723]
We are not currently depending on the latest version of `rust-bitcoinconsensus`.

Found with: https://github.com/rust-bitcoin/rust-bitcoin-maintainer-tools/pull/2

**Comment by jamillambert** [comment]
This could be done in part with PR #2833.  But further changes are be needed to implement taproot features in the verify functions in `bitcoin/src/consensus/validation.rs`

**Comment by tcharding** [comment]
Woops, sorry @jamillambert that is why I don't like suggesting issues to people, its easy to overlook some complexity when reading the issue (especially when its as poorly described as this one :)

**Comment by Kixunil** [comment]
I don't understand what "silently validate" and "correctly validate" means.

**Comment by tcharding** [comment]
By "silently validate" I meant that currently all taproot transactions are considered valid. By "correctly validate" I mean actually do some validation on them.

**Comment by benma** [comment]
Fyi, libbitcoinconsensus does not exist anymore :(

https://github.com/bitcoin/bitcoin/blob/master/doc/release-notes/release-notes-27.0.md?plain=1#L40-L53

The latest version could still be used to validate Taproot, but it is not maintained anymore unfortunately.

**Comment by apoelstra** [comment]
Yep, thanks -- we had some discussion about this at https://github.com/bitcoin/bitcoin/issues/28779
